spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/oauth2/resource/servlet/OAuth2ResourceServerJwkConfiguration.java

```diff
@@ -44,4 +47,12 @@ public JwtDecoder jwtDecoder() {
 		return new NimbusJwtDecoderJwkSupport(this.properties.getJwt().getJwkSetUri());
 	}
 
+	@Bean
+	@ConditionalOnProperty(name = "spring.security.oauth2.resourceserver.jwt.oidc-issuer-location")
```

Comment:
I think we should make it explicit that `jwk-set-uri` should take precedence by adding a custom condition that checks that `spring.security.oauth2.resourceserver.jwt.oidc-issuer-location` is present and `spring.security.oauth2.resourceserver.jwt.jwk-set-uri` is not present.

Reply:
ok, I'll add custom condition